widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java

```diff
@@ -571,34 +571,41 @@ private void updateNewMessageOperatorStatusView(String operatorProfileImgUrl) {
     }
 
     private void setupDialogCallback() {
-        dialogCallback = dialogsState -> {
-            if (dialogsState instanceof DialogsState.NoDialog) {
-                post(() -> {
-                    if (alertDialog != null) {
-                        alertDialog.dismiss();
-                        alertDialog = null;
-                    }
-                });
-            } else if (dialogsState instanceof DialogsState.UnexpectedErrorDialog) {
-                post(this::showUnexpectedErrorDialog);
-            } else if (dialogsState instanceof DialogsState.ExitQueueDialog) {
-                post(this::showExitQueueDialog);
-            } else if (dialogsState instanceof DialogsState.OverlayPermissionsDialog) {
-                post(this::showOverlayPermissionsDialog);
-            } else if (dialogsState instanceof DialogsState.EndEngagementDialog) {
-                post(() -> showEndEngagementDialog(((DialogsState.EndEngagementDialog) dialogsState).operatorName));
-            } else if (dialogsState instanceof DialogsState.UpgradeDialog) {
-                post(() -> showUpgradeDialog(((DialogsState.UpgradeDialog) dialogsState).type));
-            } else if (dialogsState instanceof DialogsState.NoMoreOperatorsDialog) {
-                post(this::showNoMoreOperatorsAvailableDialog);
-            } else if (dialogsState instanceof DialogsState.EngagementEndedDialog) {
-                post(this::showEngagementEndedDialog);
-            } else if (dialogsState instanceof DialogsState.StartScreenSharingDialog) {
-                post(this::showScreenSharingDialog);
-            } else if (dialogsState instanceof DialogsState.EnableNotificationChannelDialog) {
-                post(this::showAllowNotificationsDialog);
-            } else if (dialogsState instanceof DialogsState.EnableScreenSharingNotificationsAndStartSharingDialog) {
-                post(this::showAllowScreenSharingNotificationsAndStartSharingDialog);
+        dialogCallback = dialogState -> {
```

Comment:
I don't like those code duplications in ChatView and CallView. Maybe we should refactor it in the future.

Reply:
> I don't like those code duplications in ChatView and CallView. Maybe we should refactor it in the future.

I'll try to refactor tomorrow.